bundles/core/org.openhab.core.jsr223/src/org/openhab/core/jsr223/internal/engine/scriptmanager/ScriptManager.java

```diff
@@ -0,0 +1,189 @@
+package org.openhab.core.jsr223.internal.engine.scriptmanager;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+import javax.script.ScriptEngineFactory;
+import javax.script.ScriptEngineManager;
+import javax.script.ScriptException;
+
+import org.openhab.config.core.ConfigDispatcher;
+import org.openhab.core.items.ItemRegistry;
+import org.openhab.core.jsr223.internal.engine.RuleTriggerManager;
+import org.openhab.core.jsr223.internal.shared.Event;
+import org.openhab.core.jsr223.internal.shared.EventTrigger;
+import org.openhab.core.jsr223.internal.shared.Rule;
+import org.openhab.core.jsr223.internal.shared.StartupTrigger;
+import org.openhab.core.jsr223.internal.shared.TriggerType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Main component of script engine. It checks for available Script engines, loads scripts from the jsr_scripts directory
+ * and listens for script changes (which lead to script reloading)
+ * 
+ * @author Simon Merschjohann
+ * @since 1.7.0
+ */
+public class ScriptManager {
+	static private final Logger logger = LoggerFactory.getLogger(ScriptManager.class);
+
+	public HashMap<String, Script> scripts = new HashMap<String, Script>();
+	public HashMap<Rule, Script> ruleMap = new HashMap<Rule, Script>();
+
+	private ItemRegistry itemRegistry;
+
+	private RuleTriggerManager triggerManager;
+
+	private Thread scriptUpdateWatcher;
+
+	private static ScriptManager instance;
+
+	public ScriptManager(RuleTriggerManager triggerManager, ItemRegistry itemRegistry) {
+		this.triggerManager = triggerManager;
+		instance = this;
+		logger.info("Available engines:");
+		{
+			for (ScriptEngineFactory f : new ScriptEngineManager().getEngineFactories()) {
+				logger.info(f.getEngineName());
+			}
+		}
+
+		this.setItemRegistry(itemRegistry);
+
+		File folder = getFolder("jsr_scripts");
```

Comment:
why introducing a new folder. You could also use the existing scripts folder since the FolderObserver should only be activated for files with "rules" file extension